File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
bool mycomp(vector<int>&a,vector<int>&b)
{
    if(a[1]!=b[1])
        return a[1]<b[1];
    return a[0]<b[0];
}
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(),points.end(),mycomp);
        int arrowCnt=0,size=points.size();
        for(int i=0;i<size;i++)
        {
            int y=points[i][1];
            int iter=i;
            while(iter<size&&y>=points[iter][0]&&y<=points[iter][1])
                iter++;
            iter--;
            i=iter;
            arrowCnt++;
        }
        return arrowCnt;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points,(int[] a,int[] b) ->{
            if(a[1]!=b[1])
                return Integer.compare(a[1],b[1]);
            return Integer.compare(a[0],b[0]);
        });
        long lastTrigger=-3000000000l;
        int arrows=0;
        for(int i=0;i<points.length;i++)
        {
            if(lastTrigger<points[i][0])
            {
                arrows++;
                lastTrigger=points[i][1];
            }
        }
        return arrows;
    }
}